Source/Mosa.DeviceSystem/Bitmap.cs

```diff
@@ -20,25 +20,25 @@ public static Image CreateImage(byte[] data)
 			var height = stream.Read32(0x16);
 			var dataSectionOffset = stream.Read32(0xA);
 
-			var image = new Image(width, height);
+			var image = new Image(width, height, bytesPerPixel);
 
-			int x = 0;
-			int y = height - 1;
-
-			for (var i = dataSectionOffset; i < dataSectionOffset + (width * height * bpp); i += bpp)
+			// TODO: Support for other color depths
+			var y = height - 1;
+			do
 			{
-				var color = (bpp == 4) ? stream.Read32(i) : (int)(stream.Read24(i) | 0xFF000000); // 32 & 24 bit
+				for (uint x = 0; x < width - 1; x++)
+				{
+					dataSectionOffset += bytesPerPixel;
 
-				image.SetColor(x, y, color);
+					var color = bytesPerPixel == 4 ? stream.Read32(dataSectionOffset) : stream.Read24(dataSectionOffset) | 0xFF000000;
+					image.SetColor(x, y, color);
+				}
 
-				x++;
+				if (bytesPerPixel == 3 && width * bytesPerPixel % 4 > 0)
+					dataSectionOffset += 4;
 
-				if (x == width)
-				{
-					x = 0;
-					y--;
-				}
-			}
+				y--;
```

Comment:
move this to be the first line in the do section

Reply:
What would that change?